docs/scraping-service.md

```diff
@@ -0,0 +1,125 @@
+# Scraping Service Mode
+
+Scraping Service Mode is a third operational mode of the Grafana Cloud Agent
+that allows for clustering a set of Agent processes and distributing scrape load
+between them.
+
+Determining what to scrape is done by storing instance configuration files to an
+[API](./api.md) which then stores the configuration files in a KV store backend.
+All agents in the cluster must use the same KV store so they read the same set
+of config files.
+
+The "instance" refers to the process of:
+
+- Service discovery for all `scrape_configs` within that loaded config
+- Scrapes metrics from all discovered targets
+- Stores data in its own Write-Ahead Log specific to the loaded config
+- Remote Writes scraped metrics to the configured `remote_write` clients
+  specified within the loaded config.
+
+
+The "instance configuration file," then, is the configuration file that
+specifies the set of `scrape_configs` and `remote_write` endpoints. Each process
+of the Grafana Cloud Agent can be running multiple independent instances at
+once.
+
+When Scraping Service Mode is enabled, Agents disallow specifying
+instance configurations locally in the configuration file; using the KV store
+is required.
+
+GitOps-friendly tooling is planned to automatically load instance configuration
+files
+
+## Distributed Hash Ring
+
+Scraping Service Mode uses a Distributed Hash Ring (commonly just called "the
+ring") to cluster agents and to shard configurations within that ring. Each
+Agent joins the ring with a random distinct set of _tokens_ that are used for
+sharding. The default number of generated tokens is 128.
+
+The Distributed Hash Ring is also stored in a KV store. Since a KV store is
+also needed for storing configuration files, it is common practice to re-use
+that same KV store for the ring.
+
+When sharding, the Agent currently uses the entire contents of a config file
+stored in the KV store for load distribution. The hash of the config file is
+used as the _key_ in the ring and looks up an Agent that should be responsible
+for that config. "Price is Right" rules are used for the Agent lookup; the Agent
```

Comment:
This "looks up an Agent that should be responsible", it's not clear who/which process is doing the lookup here